client/extensions/woocommerce/app/order/order-details/table.js

```diff
@@ -136,9 +159,34 @@ class OrderDetailsTable extends Component {
 		);
 	};
 
+	renderDeleteButton = ( item, type ) => {
+		const { isEditing, translate } = this.props;
+		if ( ! isEditing ) {
+			return null;
+		}
+		return (
+			<TableItem className="order-details__item-delete">
+				<Button
+					compact
+					borderless
+					icon
+					aria-label={ translate( 'Remove %(product)s from this order', {
```

Comment:
Since the item could be a product or something like a fee, how about instead of using `%(product)s` as the string-to-be-replaced but something more like `%(itemName)s` ?